### DISCUSSION TITLE
The app no longer receives data

### DISCUSSION THREAD

**Discussion by B0bbyD0llar** [spatie/ray#346] [Q&A]
**Describe the bug**
The app no longer receives data.

**Versions**
- ray app 1.13.0.0
- spatie/ray package version: 1.21.1
- PHP version: 7.4.15

**To Reproduce**
In my test.php i have a simple call like ray()->phpinfo(); but nothing appears at the app. It doesn't really matter what command I call, even a simple ray('Hello World'); won't work.
This happens only on one of my two Win10 machines. There is no error message. A few days ago it worked perfectly. I have nothing changed at my workstation.

**Expected behavior**
Usually shows php information

**Desktop (please complete the following information):**
 - OS: Win10 20H2 - 19042.685
 - locally installed php / apache (with xampp from apachefriends)
 
 **Additional context**
It does not matter in which project I use ray. Even a completely fresh project produces the same behavior.

**Comment by patinthehat** [comment]
@B0bbyD0llar Everything on my end is working, even with a fresh project.  Would you be able to provide a small sample project that reproduces your issue?

Just to cover the bases:
- is it possible you have a firewall blocking access to port 23517 on the machine in question?
- is it possible that Ray is running on a different port?

**Comment by B0bbyD0llar** [comment]
this is my very simple test code in a fresh and empty project (only ray is required)
```php
<?php

declare(strict_types=1);

require 'vendor/autoload.php';

echo 'Hello World';
ray('Hello World');
```
As already said, it does not matter which project I take. The firewall was the first thing I checked because it works fine on my other PC. Can I test if sending the payload works without errors?

**Comment by patinthehat** [comment]
>this is my very simple test code in a fresh and empty project (only ray is required)

Can you share your `composer.json` and `ray.php` files as well?

**Comment by B0bbyD0llar** [comment]
my composer.json look like:
```json
{
    "name": "test/test",
    "description": "for testing",
    "type": "project",
    "require-dev": {
        "spatie/ray": "^1.21"
    },
    "require": {}
}
```

i dont have any ray.php files in my testing project. there is just an 'vendor' folder, composer.json, composer.lock and raytest.php

**Comment by patinthehat** [comment]
That project configuration works just fine for me.  I'm guessing the issue is something with your specific setup.

> Can I test if sending the payload works without errors?

I'm not sure what you mean by this.

I would suggest going into the Ray preferences and changing the port number to `23516`, and creating a `ray.php` file in the root directory of your project:

```php
<?php
// ray.php

return [
    'enable' => true,
    'host' => '127.0.0.1',
    'port' => 23516,
    'remote_path' => null,
    'local_path' => null,
    'always_send_raw_values' => false,
];
```

Trying this will eliminate the possibility that there's a conflict with the default port.  _(Don't forget to change the port back to `23517` in Ray if this doesn't work.)_

**Comment by B0bbyD0llar** [comment]
> I'm not sure what you mean by this.

I mean to test if the PHP script is giving out the data properly and the app is not receiving it correctly. 

I have tried different ports. Unsuccessfully.

**Comment by patinthehat** [comment]
I'm not running Windows, so I'm not sure what else to do beyond that.  Hopefully one of the Windows users can give you some guidance.

> I mean to test if the PHP script is giving out the data properly and the app is not receiving it correctly.

To do this, I use a custom http proxy written in nodejs to inspect the data that's being sent.  I'd be happy to share it if you'd like to debug the data being sent from PHP to the Ray app.  I'm not sure it'd be much help in this scenario, but let me know if you'd like to give it a shot.

**Comment by patinthehat** [comment]
@B0bbyD0llar were you able to resolve this issue? If not, try taking a look at [ray-proxy](https://github.com/permafrost-dev/ray-proxy) to see if you can debug the data being sent (or not) from your code to the Ray app.

**Reply by B0bbyD0llar** [reply]
Unfortunately, my issue was closed without solving the problem. I will try to get more information with the proxy. Thanks for the tip.

**Reply by patinthehat** [reply]
If you need any help with it, please feel free to open an issue in the ray-proxy repo.
